File: communications/comms.py
# The python half of the json test
import json
import time

def send(message, filename="tocpp"):
    print "in send"
    with open(filename, 'w') as tocpp:
        print "file opened"
        tocpp.write(message)

def recieve(filename="topy"):
    with open(filename, 'r') as topy:
        message = topy.read()
    return message

if __name__ == '__main__':
    # Emulate the autoencoder interaction
    # Read, think, write
    print "getting message"
    message = recieve()
    print "recieved ", message
    print "sleeping..."
    time.sleep(1)
    send(message)
    print "Message sent"

File: communications/comms.cpp
/* The C++ 11 half of the json test, using nlohman/json as the C++ json library */
#include "nlohmann/json.hpp"
#include <iostream>
#include <fstream>
using json= nlohmann::json;

void send(const std::string & message, const std::string & filename)
{
	std::ofstream topy(filename);
	topy << message;
	topy.close();
}

std::string recieve(const std::string &filename)
{
	std::string message;
	std::cout << "Opening file... " << std::endl;
	std::ifstream tocpp(filename);
	std::cout << "Opened!" << std::endl;
	tocpp >> message;
	std::cout << "Finished writing" << std::endl;
	tocpp.close();
	std::cout << " and closed" << std::endl;
	return message;
}

